bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1295,4 +1298,127 @@ public boolean equals(Object otherPair) {
             return true;
         }
     }
+
+    /**
+     * Checks whether a given {@link Byte} has an implicit initial value or not.
+     * @param type {@link Byte} to be analyzed.
+     * @return whether there's an implicit initial value or not.
+     */
+    public static boolean hasImplicitInitialValue(BType type) {
+        return hasImplicitInitialValue(type, new ArrayList<>());
+    }
+
+    private static boolean hasImplicitInitialValue(BType type, List<BType> unAnalyzedTypes) {
+        if (type == null) {
```

Comment:
Ideally, type shouldn't be null right? Do we know where they become null? Is it due to array types being set incorrectly?